HashRegistrarSimplified.sol

```diff
@@ -118,19 +118,58 @@ contract Registrar {
     event HashInvalidated(bytes32 indexed hash, string indexed name, uint value, uint now);
 
     struct entry {
-        Mode status;
         Deed deed;
         uint registrationDate;
         uint value;
         uint highestBid;
     }
 
+    // State transitions for names:
+    //   Open -> Auction (startAuction)
+    //   Auction -> Reveal
+    //   Reveal -> Owned
+    //   Reveal -> Open (if nobody bid)
+    //   Owned -> Forbidden (invalidateName)
```

Comment:
Doing so complicates the bidding process a lot - so with this PR you can only forbid auctions that have finished. We could potentially add an Open -> Forbidden transition too.